ocs_ci/deployment/flexy.py

```diff
@@ -0,0 +1,274 @@
+"""
+All the flexy related classes and functionality lives here
+"""
+import logging
+import os
+import sys
+
+import io
+import configparser
+import subprocess
+from concurrent.futures import ThreadPoolExecutor
+
+from ocs_ci.framework import config, merge_dict
+from ocs_ci.ocs import constants
+from ocs_ci.utility.utils import (
+    get_ocp_version, TimeoutSampler, clone_repo
+)
+from ocs_ci.ocs import exceptions
+
+logger = logging.getLogger(__name__)
+
+
+class FlexyBase(object):
+    """
+    A base class for all types of flexy installs
+    """
+    def __init__(self):
+        # Host dir path which will be mounted inside flexy container
+        # This will be root for all flexy housekeeping
+        self.cluster_name = config.ENV_DATA['cluster_name']
+        self.flexy_mnt_host_dir = os.path.expanduser(
+            config.ENV_DATA['flexy_mnt_host_dir']
+        )
+        # Host dir path for private_conf dir
+        self.flexy_host_private_conf_dir_path = os.path.join(
+            self.flexy_mnt_host_dir, config.ENV_DATA['flexy_host_private_conf_dir']
+        )
```

Comment:
ack, Initial idea was to by default download private-conf and then make the changes and mount it inside container .  I will default this behaviour to just use a default env file so that no clone is necessary by us instad flexy does it itself inside container

Reply:
I think, that if you would like to use the env file from the private repository, than you will have to clone it. In other words, if my understanding is correct, the env file have to be accessible to the `podman run ...` command before it even start to clone any repositories inside of the container.